src/main/java/com/redhat/cloud/notifications/routers/BehaviorGroupMigrationService.java

```diff
@@ -0,0 +1,231 @@
+public class BehaviorGroupMigrationService {
+
+    public static final String CONFIRMATION_TOKEN = "ready-set-go";
+
+    private static final Logger LOGGER = Logger.getLogger(BehaviorGroupMigrationService.class.getName());
+
+    @Inject
+    Mutiny.Session session;
+
+    @GET
+    public Uni<Response> migrate(@QueryParam("confirmation-token") String confirmationToken) {
+        if (!CONFIRMATION_TOKEN.equals(confirmationToken)) {
+            throw new BadRequestException("You forgot to confirm the migration!");
+        }
+        LOGGER.debug("[BG MIGRATION] Start");
+        LOGGER.debug("[BG MIGRATION] Migrating global default actions");
+        // All DB changes will be rolled-back if anything goes wrong.
+        // TODO Test it!
+        return session.withTransaction(tx -> {
+            /*
+             * The current default actions are not bound to a specific bundle, so we have to create for each existing
+             * bundle a default behavior group that will contain the current default actions.
+             */
+            return session.createQuery("FROM Bundle", Bundle.class)
+                    .getResultList()
+                    .onItem().invoke(bundles -> LOGGER.debugf("[BG MIGRATION] __ %d bundle(s) found", bundles.size()))
+                    .onItem().transformToMulti(Multi.createFrom()::iterable)
+                    // Let's iterate over all accounts that saved default actions.
+                    .onItem().transformToUniAndConcatenate(bundle -> {
+                        LOGGER.debugf("[BG MIGRATION] __ Migrating global default actions for Bundle[id=%s, name=%s]", bundle.getId(), bundle.getName());
+                        return session.createQuery("SELECT DISTINCT id.accountId FROM EndpointDefault", String.class)
+                                .getResultList()
+                                .onItem().invoke(accounts -> LOGGER.debugf("[BG MIGRATION] ____ %d account(s) with global default actions found", accounts.size()))
+                                .onItem().transformToMulti(Multi.createFrom()::iterable)
+                                .onItem().transformToUniAndConcatenate(accountId -> {
+                                    LOGGER.debugf("[BG MIGRATION] ____ Transforming global default actions into default behavior group for account '%s'", accountId);
+                                    // The current account has default actions so it needs a default behavior group.
+                                    BehaviorGroup defaultGroup = new BehaviorGroup();
+                                    defaultGroup.setAccountId(accountId);
+                                    defaultGroup.setDisplayName("Default");
+                                    defaultGroup.setBundle(bundle);
+                                    defaultGroup.setBundleId(bundle.getId()); // Yes, we need to set that even if the bundle was already set.
+                                    defaultGroup.setDefaultBehavior(true);
```

Comment:
What happens with accounts that don't have any default set ? They won't get a default behavior group?

Also, I'm wondering if we really need a default behavior group for the feature-parity. I was under the impression that the bundle-defaults would do and that is OK to not have any default behavior group managed by the user.

Reply:
> What happens with accounts that don't have any default set ? They won't get a default behavior group?

Yes. A default behavior group will be created only if an account _saved_ default endpoints (in the `endpoint_defaults` table) before the migration. The default endpoints may or may not be linked with event types (in the `endpoint_targets` table), it doesn't have an influence on the default behavior group creation (it will be created anyway).

Reply:
I created #262 to get rid of account-level behavior groups. I'll rebase this PR and simplify the migration after the merge.

Reply:
I changed a lot of things, this discussion is no longer relevant.